sdks/python/apache_beam/internal/gcp/auth.py

```diff
@@ -32,6 +33,10 @@
 except ImportError:
   GceAssertionCredentials = None
 
+# apitools use urllib with the global timeout. Set it to 60 seconds
+# to prevent network related stuckness issues.
+socket.setdefaulttimeout(60)
+  
```

Comment:
The unknown part here is whether this gets applied on a remote worker. The defaulttimeout variable is a CPython implementation detail, so I don't believe it gets pickled and passed to workers. Will this module get imported on workers, or will the pickled version of it be used (such as with save_main_session)?

My notes are here:
https://issues.apache.org/jira/browse/BEAM-7616?focusedCommentId=16913549&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16913549

Reply:
I am not sure. Adding a log statement here. I should be able to run a job and look and figure out from the logs.